File: LeetCode/0001.two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer,Integer>map = new HashMap<>(nums.length);
        for (int i = 0; i < nums.length; i++)   map.put(nums[i], i);
        for (int i = 0; i < nums.length; i++){
            if (map.get(target - nums[i]) != null && i != map.get(target - nums[i])){
                res[0] = map.get(target - nums[i]);
                res[1] = i;
                break;
            }
        }
        return res;
    }
}

File: LeetCode/0001.two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val res = IntArray(2)
        val map = mutableMapOf<Int, Int>()
        for (i in nums.indices){
            map[nums[i]] = i
        }
        for (i in nums.indices){
            if (map[target - nums[i]] != null && i != map[target - nums[i]]){
                res[0] = map[target - nums[i]]!!
                res[1] = i
                break
            }
        }
        return res
    }
}